openaerostruct/geometry/utils.py

```diff
@@ -396,23 +397,18 @@ def gen_rect_mesh(num_x, num_y, span, chord, span_cos_spacing=0.0, chord_cos_spa
         half_wing = cosine * span_cos_spacing + (1 - span_cos_spacing) * uniform
         full_wing = np.hstack((-half_wing[:-1], half_wing[::-1])) * span
 
-    nx2 = (num_x + 1) // 2
-    beta = np.linspace(0, np.pi / 2, nx2)
-
-    # mixed spacing with span_cos_spacing as a weighting factor
-    # this is for the chordwise spacing
-    cosine = 0.5 * np.cos(beta)  # cosine spacing
-    uniform = np.linspace(0, 0.5, nx2)[::-1]  # uniform spacing
-    half_wing = cosine * chord_cos_spacing + (1 - chord_cos_spacing) * uniform
-    full_wing_x = np.hstack((-half_wing[:-1], half_wing[::-1])) * chord
-
-    # Special case if there are only 2 chordwise nodes
-    if num_x <= 2:
-        full_wing_x = np.array([0.0, chord])
+    # --- chordwise discretization ---
+    cosine = -0.5 + 0.5 * (1 - np.cos(np.linspace(0, np.pi, num_x)))  # cosine spacing from -0.5 to 0.5
+    uniform = np.linspace(-0.5, 0.5, num_x)  # uniform spacing
+    # mixed spacing with chord_cos_spacing as a weighting factor
+    wing_x = cosine * chord_cos_spacing + (1 - chord_cos_spacing) * uniform  # from -0.5 to 0.5
+    wing_x += 0.5  # offset by 0.5 so it goes 0 to 1 (leading edge = 0, trailing edge = 1)

```

Comment:
Minor, but why include this offset by 0.5 line after already shifting the cosine and uniform distributions by -0.5? I think may as well just have them both go from 0 to 1 and get rid of this line. As far as I can tell, they should give the same result.

Reply:
Good catch, fixed